src/interpreter/s2s_reverse_mode_ad.jl

```diff
@@ -368,7 +369,7 @@ function make_ad_stmts!(stmt::PiNode, line::ID, info::ADInfo)
     # Assemble the above lines and construct reverse-pass.
     return ad_stmt_info(
         line,
-        PiNode(stmt.val, fcodual_type(_type(stmt.typ))),
+        PiNode(__inc(stmt.val), fcodual_type(_type(stmt.typ))),
```

Comment:
@yebai it's this line.

If the first element of `PiNode` is `Argument(2)` in the primal code, then it needs to correspond to `Argument(3)` in the forwards-pass code -- this is a consistent pattern that you'll see across the codebase (argument `n` in the primal is argument `n+1` in the forwards-pass -- this is handled by calls to the internal `__inc` function in various places).

I had previously only tested cases where the first argument to a `PiNode` was an `SSA`, rather than an `Argument` -- it looks turns out that seeing an `Argument` in a `PiNode` is fairly rare in practice.

The test that function `arg_in_pi_node` that I've added in `src/test_utils.jl` is a minimal example in which this happens.

This is definitely one of those bugs that took me a while to understand, but for which the fix is entirely trivial.